overlord/snapstate/backend/copydata_test.go

```diff
@@ -150,20 +150,28 @@ func (s *copydataSuite) TestCopyDataNoUserHomes(c *C) {
 }
 
 func (s *copydataSuite) populateData(c *C, revision snap.Revision) {
-	datadir := filepath.Join(dirs.SnapDataDir, "hello/"+revision.String())
+	datadir := filepath.Join(dirs.SnapDataDir, "hello", revision.String())
 	subdir := filepath.Join(datadir, "random-subdir")
 	err := os.MkdirAll(subdir, 0755)
 	c.Assert(err, IsNil)
-	err = ioutil.WriteFile(filepath.Join(subdir, "canary"), nil, 0644)
+	err = ioutil.WriteFile(filepath.Join(subdir, "canary"), []byte(fmt.Sprintln(revision)), 0644)
 	c.Assert(err, IsNil)
 }
 
+func (s *copydataSuite) populatedData(d string) string {
+	bs, err := ioutil.ReadFile(filepath.Join(dirs.SnapDataDir, "hello", d, "random-subdir", "canary"))
+	if err != nil {
```

Comment:
bit of paranoia but this should probably check for IsErrNoExist and panic otherwise?